# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.h
# Commit: ac04b727a98b
# Full Hash: ac04b727a98bb1d4e0644665b59ed4da3e9e3858
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P8: Move GetGroupPosition to Accessible. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134211
# ==============================================================================

diff -r 5d76535b47fc -r ac04b727a98b accessible/basetypes/Accessible.h
--- a/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:37 2022 +0000
+++ b/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:37 2022 +0000
@@ -52,6 +52,19 @@
   eNameFromTooltip
 };
 
+/**
+ * Group position (level, position in set and set size).
+ */
+struct GroupPos {
+  GroupPos() : level(0), posInSet(0), setSize(0) {}
+  GroupPos(int32_t aLevel, int32_t aPosInSet, int32_t aSetSize)
+      : level(aLevel), posInSet(aPosInSet), setSize(aSetSize) {}
+
+  int32_t level;
+  int32_t posInSet;
+  int32_t setSize;
+};
+
 class Accessible {
  protected:
   Accessible();
@@ -113,6 +126,11 @@
   bool HasGenericType(AccGenericType aType) const;
 
   /**
+   * Return group position (level, position in set and set size).
+   */
+  virtual GroupPos GroupPosition();
+
+  /**
    * Return embedded accessible child at the given index.
    */
   virtual Accessible* EmbeddedChildAt(uint32_t aIndex) = 0;